docs/iris/src/userguide/loading_iris_cubes.rst

```diff
@@ -314,21 +315,53 @@ for many years:
 
 .. doctest:: timeseries_range
     :options: +NORMALIZE_WHITESPACE, +ELLIPSIS
-    
+
     >>> print(long_ts.coord('time'))
     DimCoord([2007-04-09 00:00:00, 2007-04-16 00:00:00, 2007-04-23 00:00:00,
               ...
               2010-02-01 00:00:00, 2010-02-08 00:00:00, 2010-02-15 00:00:00],
              standard_name='time', calendar='gregorian')
 
-We can select points within a certain part of the year, in this case between
-the 15th of July through to the 25th of August, by combining the datetime cell
-functionality with PartialDateTime:
+Given two dates in datetime format, we can select all points between them.
+
+.. doctest:: timeseries_range
+    :options: +NORMALIZE_WHITESPACE, +ELLIPSIS
+
+    >>> d1 = datetime.datetime.strptime('20070715T0000Z', '%Y%m%dT%H%MZ')
+    >>> d2 = datetime.datetime.strptime('20070825T0000Z', '%Y%m%dT%H%MZ')
+    >>> st_swithuns_daterange_07 = iris.Constraint(
+    ...     time=lambda cell: d1 <= cell.point < d2)
+    >>> within_st_swithuns_07 = long_ts.extract(st_swithuns_daterange_07)
+    >>> print(within_st_swithuns_07.coord('time'))
+    DimCoord([2007-07-16 00:00:00, 2007-07-23 00:00:00, 2007-07-30 00:00:00,
+              2007-08-06 00:00:00, 2007-08-13 00:00:00, 2007-08-20 00:00:00],
+             standard_name='time', calendar='gregorian')
+
+Alternatively, we may do the same thing with :class:`iris.time.PartialDateTime`
+objects.
+
+.. doctest:: timeseries_range
+    :options: +NORMALIZE_WHITESPACE, +ELLIPSIS
+
+    >>> pdt1 = PartialDateTime(year=2007, month=7, day=15)
+    >>> pdt2 = PartialDateTime(year=2007, month=8, day=25)
+    >>> st_swithuns_daterange_07 = iris.Constraint(
+    ...     time=lambda cell: pdt1 <= cell.point < pdt2)
+    >>> within_st_swithuns_07 = long_ts.extract(st_swithuns_daterange_07)
+    >>> print(within_st_swithuns_07.coord('time'))
+    DimCoord([2007-07-16 00:00:00, 2007-07-23 00:00:00, 2007-07-30 00:00:00,
+              2007-08-06 00:00:00, 2007-08-13 00:00:00, 2007-08-20 00:00:00],
+             standard_name='time', calendar='gregorian')
+
+A more complex example might require selecting points over an annually repeating
+date range. We can select points within a certain part of the year, in this case
+between the 15th of July through to the 25th of August. By making use of
+PartialDateTime this becomes simple:
 
 .. doctest:: timeseries_range
 
     >>> st_swithuns_daterange = iris.Constraint(
-    ...     time=lambda cell: PartialDateTime(month=7, day=15) < cell < PartialDateTime(month=8, day=25))
+    ...     time=lambda cell: PartialDateTime(month=7, day=15) <= cell < PartialDateTime(month=8, day=25))
```

Comment:
Why this change?

Reply:
So there's two reasons for this. The first is that I wanted to make it more obvious that PartialDateTime range was the same as the range defined by datetetime objects. The datetime objects measure time from the start of the day, so for them it is necessary to specify the first date in the range and the first date outside of the range. With PartialDateTime, this is not necessary but I made the change so that the dates matched up.
The other reason is that this example appears to be a reference to a proverb about St Swithun's day https://en.wikipedia.org/wiki/Swithun. I decided that an object called `st_swithuns_daterange` could reasonably be expected to contain St Swithun's day itself (this does mean that there are now 41 days in the daterange, though this still fits with some readings of the proverb).